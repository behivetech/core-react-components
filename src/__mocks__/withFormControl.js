export default (WrappedComponent) => function FormFieldControlled(props) { 
    const formState = {
        formDisabled: false,
        initialSubmit: false,
        getFieldError: jest.fn(),
        getFieldValue: jest.fn(),
        setFieldValidators: jest.fn(),
        setFieldValue: jest.fn(),
        setFieldValueDebounced: jest.fn(),
        shouldDisable: jest.fn(),
    };

    return <WrappedComponent {...props} formState={formState} />; 
};
