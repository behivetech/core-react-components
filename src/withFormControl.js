// Vendor Libs
import React, {Component} from 'react';
import PropTypes from 'prop-types';

// Components
import {FormContext} from './FormControlled';

export default function (WrappedComponent) {
    class FormFieldControlled extends Component {
        constructor(props) {
            super(props);
            this.renderChild = this.renderChild.bind(this);
        }

        renderChild(formState) {
            return (
                <WrappedComponent
                    {...this.props}
                    formState={formState}
                    disabled={formState.formDisabled || this.props.disabled}
                />
            );
        }

        render() {
            return (
                <FormContext.Consumer>
                    {this.renderChild}
                </FormContext.Consumer>
            );
        }
    }

    FormFieldControlled.propTypes = {
        /** disables the field */
        disabled: PropTypes.bool,
    };

    return FormFieldControlled;
}

