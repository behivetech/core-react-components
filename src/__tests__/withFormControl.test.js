import withFormControl from '../withFormControl';
jest.mock('../Form');

describe('withFormControl', () => {
    const MockComponentWrapped = (props) => <div />;
    const MockComponentControlled = withFormControl(MockComponentWrapped);
    // eslint-disable-next-line
    const MockComponent = (props) => <MockComponentControlled {...props} />;

    afterEach(() => {
        jest.clearAllMocks();
    });

    it('should pass fromState to the warapped component', () => {
        const wrapper = mount(<MockComponent name="mockName" className="mock-class" />);

        expect(wrapper).toMatchSnapshot();
    });
});