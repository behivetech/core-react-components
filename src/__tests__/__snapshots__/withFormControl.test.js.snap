// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withFormControl should pass fromState to the warapped component 1`] = `
<MockComponent
  className="mock-class"
  name="mockName"
>
  <FormFieldControlled
    className="mock-class"
    name="mockName"
  >
    <MockComponentWrapped
      className="mock-class"
      formState={
        Object {
          "formDisabled": false,
          "getFieldValue": [Function],
          "getFieldValues": [Function],
          "initialSubmit": false,
          "setFieldError": [Function],
          "setFieldValue": [Function],
          "setFieldValueDebounced": [Function],
          "unsetFieldError": [Function],
        }
      }
      name="mockName"
    >
      <div />
    </MockComponentWrapped>
  </FormFieldControlled>
</MockComponent>
`;
